### DISCUSSION TITLE
Is it possible to add dataChannel on this code?

### DISCUSSION THREAD

**Discussion by siyeol** [Dirvann/webrtc-video-conference-example-simple-peer#11] [Q&A]
Hello,
I am thankfully using this code on my project.
However, I need to add datachannel so that webRTC could share JSON..
Is there any way I could add those functions?

Can I just add these codes
const peerConnection = new RTCPeerConnection(configuration);
const dataChannel = peerConnection.createDataChannel();
on public/js/main.js ??

**Comment by Dirvann** [comment]
Yeah you can make a PR if you want

**Reply by siyeol** [reply]
I made a PR! Plz check

**Comment by siyeol** [comment]
Dear Mr.Vanberveren
 
 
Hello, I am Siyeol Choi who asked for availability of additional function.
 
I made a pull request about adding a Text Chat on your code.
 
I did not use dataChannel since your code does signaling with Socket.io, 
 
so I thought it would be more advantageous to use Socket.io rather than to make another dataChannel Connection.
 
If you don't mind could you please merge it into the main Branch?
 
 
Sincerely,
Siyeol Choi
 
 
-----Original Message-----
